test/signalrclient-e2e-tests/connection_tests.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include "stdafx.h"
+#include <string>
+#include "cpprest\details\basic_types.h"
+#include "cpprest\json.h" 
+#include "connection.h"
+#include "hub_connection.h"
+#include "console_log_writer.h"
+
+using namespace signalr;
+using namespace utility;
+using namespace std;
+
+TEST(connection_tests, send_message)
+{
+	shared_ptr<log_writer> Log_writer(make_shared<console_log_writer>());
+
+	connection conn(L"http://localhost:8081/raw-connection", L"", trace_level::all, Log_writer);
+
+	string_t message;
+	int count = 0;
+	pplx::event test_event{};
+
+	conn.set_message_received([&message, &count, &test_event](const string_t& payload)
+	{
+		message = payload;
+		wcout << "Received message: " << payload << endl;
```

Comment:
I don't think we should write anything in tests - they should even pass or fail. It should be fine to write to a log file though. Ideally using the same mechanism as the one passed when creating the connection.